triedb/pathdb/database_test.go

```diff
@@ -520,17 +524,18 @@ func TestDatabaseRecoverable(t *testing.T) {
 	}
 }
 
-// TODO(joey): fail when using asyncbuffer
-//
-//nolint:unused
-func testDisable(t *testing.T) {
+func TestDisable(t *testing.T) {
 	// Redefine the diff layer depth allowance for faster testing.
 	maxDiffLayers = 4
 	defer func() {
 		maxDiffLayers = 128
 	}()
 
 	tester := newTester(t, 0, false, 32)
+	bottom := tester.db.tree.bottom()
+	if err := bottom.buffer.flush(tester.db.diskdb, tester.db.freezer, nil, bottom.id, true); err != nil {
```

Comment:
Hi, I have a question. In actual use, are there any problems similar to UT, such as shutdown or other scenarios? Currently, the modification in UT is manually flushed once, but are there any similar problems in actual use?